client/lib/signup/step-actions.js

```diff
@@ -9,20 +9,102 @@ import async from 'async';
 /**
  * Internal dependencies
  */
+import { cartItems } from 'lib/cart-values';
 import wpcom from 'lib/wp' ;
 const sites = require( 'lib/sites-list' )();
 const user = require( 'lib/user' )();
 import { getSavedVariations } from 'lib/abtest';
 import SignupCart from 'lib/signup/cart';
+import { startFreeTrial } from 'lib/upgrades/actions';
+
+function addDomainItemsToCart( callback, dependencies, { domainItem, googleAppsCartItem, isPurchasingItem, siteUrl, themeSlug, themeItem } ) {
+	wpcom.undocumented().sitesNew( {
+		blog_name: siteUrl,
+		blog_title: siteUrl,
+		options: {
+			theme: dependencies.theme
+		},
+		validate: false,
+		find_available_url: isPurchasingItem
+	}, function( error, response ) {
+		if ( error ) {
+			callback( error );
+
```

Comment:
I think it's really rare to have callbacks return something, so just writing `return callback( err );` makes the error handling code shorter (since it's less interesting usually).
Anyway, this is javascript and those things are up to the author IMO. I think it makes sense to keep the original choice otherwise this is going to be changed each time a new developer works on it.

In this case, since it helped you understand the steps better, you have my blessing to keep your syntax :)